pandas/io/stata.py

```diff
@@ -1821,6 +1821,13 @@ def _do_convert_missing(self, data: DataFrame, convert_missing: bool) -> DataFra
         replacements = {}
         for i in range(len(data.columns)):
             fmt = self._typlist[i]
+            # missing code for double was different in version 105 and prior
+            # recode instances of this to the currently used value
+            if self._format_version <= 105 and fmt == "d":
+                data.iloc[:, i] = data.iloc[:, i].replace(
+                    float.fromhex("0x1.0p333"), self.MISSING_VALUES["d"]
```

Comment:
Nit: Could you define `float.fromhex("0x1.0p333")` outside the loop?

Reply:
Yes, that should be straightforward, I'll make that change.